test/e2e/proxy_test.go

```diff
@@ -281,30 +288,48 @@ func testProxyNodePortService(t *testing.T, isIPv6 bool) {
 	defer teardownTest(t, data)
 	skipIfProxyAllDisabled(t, data)
 
-	nodes := []string{nodeName(0), nodeName(1)}
-	nodeIPs := []string{controlPlaneNodeIPv4(), workerNodeIPv4(1)}
+	nodes := []string{nodeName(0)}
+	nodeIPs := []string{controlPlaneNodeIPv4()}
 	ipProtocol := corev1.IPv4Protocol
 	if isIPv6 {
-		nodeIPs = []string{controlPlaneNodeIPv6(), workerNodeIPv6(1)}
+		nodeIPs = []string{controlPlaneNodeIPv6()}
 		ipProtocol = corev1.IPv6Protocol
 	}
+	if clusterInfo.numNodes > 2 {
+		// For a Windows cluster, ensure a Windows Node is included.
+		nodes = append(nodes, nodeName(2))
```

Comment:
I don't see how this ensures that a Windows Node is selected, except maybe on some specific test clusters.

We have `clusterInfo.windowsNodes` to access Windows Nodes

Reply:
Updated.